lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/sys.rb

```diff
@@ -71,13 +71,13 @@ class Console::CommandDispatcher::Stdapi::Sys
   # Options used by the 'reg' command.
   #
   @@reg_opts = Rex::Parser::Arguments.new(
-    "-d" => [ true,  "The data to store in the registry value."		   ],
-    "-h" => [ false, "Help menu."						   ],
-    "-k" => [ true,  "The registry key path (E.g. HKLM\\Software\\Foo)."	   ],
-    "-t" => [ true,  "The registry value type (E.g. REG_SZ)."		   ],
-    "-v" => [ true,  "The registry value name (E.g. Stuff)."		   ],
+    "-d" => [ true,  "The data to store in the registry value." ],
+    "-h" => [ false, "Help menu." ],
+    "-k" => [ true,  "The registry key path (E.g. HKLM\\Software\\Foo)." ],
+    "-t" => [ true,  "The registry value type (E.g. REG_SZ)." ],
+    "-v" => [ true,  "The registry value name (E.g. Stuff)." ],
     "-r" => [ true,  "The remote machine name to connect to (with current process credentials" ],
-    "-w" => [ false, "Set KEY_WOW64 flag, valid values [32|64]."		   ])
+    "-w" => [ true,  "Set KEY_WOW64 flag, valid values [32|64]." ])
```

Comment:
TODO: We recently changed the rex arguments parser, we should just double check that we didn't accidentally introduce a regression when we swapped out some of of the implementation details https://github.com/rapid7/metasploit-framework/pull/16014

Reply:
I ran the verification steps from https://github.com/rapid7/metasploit-framework/pull/16014 on this branch and the results were as expected. Let me know if there's anything else you think we should check for in terms of regression testing

Reply:
I was able to confirm that this functionality was broken before the rex parser argument changes

I confirmed this by grabbing the commit prior to the rex argument parser update, and verifying that the `-w` flag wasn't handle correctly - as it was missing `true` for requiring an argument

```
git checkout 158a0aa30cd7476b572875fd3ef81db51630e646~1
```